cmd/kubeadm/app/cmd/phases/addons.go

```diff
@@ -0,0 +1,148 @@
+	}
+
+	cmd.AddCommand(getAddonsSubCommands()...)
+	return cmd
+}
+
+// EnsureAllAddons install all addons to a Kubernetes cluster.
+func EnsureAllAddons(cfg *kubeadmapi.MasterConfiguration, client clientset.Interface) error {
+
+	addonActions := []func(cfg *kubeadmapi.MasterConfiguration, client clientset.Interface) error{
+		dnsaddon.EnsureDNSAddon,
+		proxyaddon.EnsureProxyAddon,
+	}
+
+	for _, action := range addonActions {
+		err := action(cfg, client)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// getAddonsSubCommands returns sub commands for addons phase
+func getAddonsSubCommands() []*cobra.Command {
+	cfg := &kubeadmapiext.MasterConfiguration{}
+	// Default values for the cobra help text
+	api.Scheme.Default(cfg)
+
+	var cfgPath, kubeConfigFile string
+	var subCmds []*cobra.Command
+
+	subCmdProperties := []struct {
+		use     string
+		short   string
+		cmdFunc func(cfg *kubeadmapi.MasterConfiguration, client clientset.Interface) error
+	}{
+		{
+			use:     "all",
+			short:   "Install all addons to a Kubernetes cluster",
+			cmdFunc: EnsureAllAddons,
+		},
+		{
+			use:     "kube-dns",
+			short:   "Install the kube-dns addon to a Kubernetes cluster.",
+			cmdFunc: dnsaddon.EnsureDNSAddon,
+		},
+		{
+			use:     "kube-proxy",
+			short:   "Install the kube-proxy addon to a Kubernetes cluster.",
+			cmdFunc: proxyaddon.EnsureProxyAddon,
+		},
+	}
+
+	for _, properties := range subCmdProperties {
+		// Creates the UX Command
+		cmd := &cobra.Command{
+			Use:   properties.use,
+			Short: properties.short,
+			Run:   runAddonsCmdFunc(properties.cmdFunc, cfg, cfgPath),
+		}
+
+		// Add flags to the command
+		cmd.Flags().StringVar(&kubeConfigFile, "kubeconfig", "/etc/kubernetes/admin.conf", "The KubeConfig file to use for talking to the cluster")
+		cmd.Flags().StringVar(&cfgPath, "config", cfgPath, "Path to kubeadm config file (WARNING: Usage of a configuration file is experimental)")
+		cmd.Flags().StringVar(&cfg.KubernetesVersion, "kubernetes-version", cfg.KubernetesVersion, `Choose a specific Kubernetes version for the control plane.`)
+		cmd.Flags().StringVar(&cfg.ImageRepository, "image-repository", cfg.ImageRepository, `Choose a container registry to pull control plane images from.`)
+		if properties.use == "all" || properties.use == "kube-proxy" {
+			cmd.Flags().StringVar(&cfg.API.AdvertiseAddress, "apiserver-advertise-address", cfg.API.AdvertiseAddress, `The IP address the API Server will advertise it's listening on. 0.0.0.0 means the default network interface's address.`)
+			cmd.Flags().Int32Var(&cfg.API.BindPort, "apiserver-bind-port", cfg.API.BindPort, `Port for the API Server to bind to.`)
+			cmd.Flags().StringVar(&cfg.Networking.PodSubnet, "pod-network-cidr", cfg.Networking.PodSubnet, `Specify range of IP addresses for the pod network; if set, the control plane will automatically allocate CIDRs for every node.`)
+		} else if properties.use == "all" || properties.use == "kube-dns" {
```

Comment:
By using `else if`the `all`subcommand won't have the `kube-dns` flags. Please move `if properties.use == "all" || properties.use == "kube-dns" ` to a separate line

Reply:
What @fabriziopandini said.